File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> counts;
    for (const auto& cloth : clothes) {
        counts[cloth[1]]++;
    }

    int result = 1;
    for (const auto& pair : counts) {
        result *= (pair.second + 1);
    }

    return result - 1;
}

File: 프로그래머스/2/42578. 의상/의상.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int solution(string[,] clothes) 
    {
        int c = clothes.GetLength(0);
        Dictionary<string, int> dic = new Dictionary<string, int>();
        for(int i=0; i<c ;i++)
        {
            if(dic.ContainsKey(clothes[i,1]))
            {
                dic[clothes[i,1]]++;
            }
            else
            {
                dic[clothes[i,1]] = 1;
            }
        }
        
        int result = 1;
        foreach(var d in dic)
        {
            result *= (d.Value + 1);
        }
        return result - 1;
    }
}